src/drivers/net/e1000.c

```diff
@@ -283,11 +288,11 @@ static int e1000_open(struct net_device *dev) {
 	REG_ORIN(e1000_reg(dev, E1000_REG_RCTL),  E1000_REG_RCTL_MPE);
 
 	for (size_t i = 0; i < E1000_RXDESC_NR; i ++) {
-	        struct sk_buff *skb;
-		skb = skb_alloc(ETH_FRAME_LEN);
-		if (skb == NULL)
-			panic("Can't allocate skb %d", i);
-		rx_skbs[i]= skb;
+	        struct sk_buff *skb = skb_alloc(E1000_MAX_RX_LEN);
+		if (skb == NULL) {
+			return -ENOMEM;
+		}
+		rx_skbs[i] = skb;
 		rx_descs[i].buffer_address = (uint32_t) skb->mac.raw;
 	}
```

Comment:
I've just realized that this shouldn't be done in `e1000_open` at all, since it initializes some *global* structures and doesn't use `dev` being opened. This should be in `e1000_init`, am I right?
Anyway, just breaking the loop and returning is not graceful enough, and one should clean up already allocated data, i.e. rollback the loop and `skb_free` each item from (i .. 0].